### PR TITLE
Hotfix/no duplicate properties

### LINKED ISSUE #280: no-duplicate-properties should allow repeat properties when they are directly next to each other
Duplicating properties is sometimes required - such as when providing fallbacks for unsupported features, e.g. the following is a legitimate usage but currently raises a warning:

```scss
.foo {
  display: inline-block;
  display: flex;
}
```

Sass-lint should follow [CSSLint's lead](https://github.com/CSSLint/csslint/wiki/Disallow-duplicate-properties) and allow duplicate properties when they have different values but are immediately next to each other. I.e. the following are warnings:

```scss
/* properties with the same value */
.mybox {
  border: 1px solid black;
  border: 1px solid black;
}

/* properties separated by another property */
.mybox {
  border: 1px solid black;
  color: green;
  border: 1px solid red;
}
```

But the following should be acceptable, and not raise a warning:

```scss
/* one after another with different values */
.mybox {
  border: 1px solid black;
  border: 1px solid red;
}
```

### PR DESCRIPTION
**Pull request by DanPurdy**
This PR updates the no-duplicate-properties rule as discussed in #280 

Excluded duplicate properties must now immediately follow each other when used.

Extra tests are added to check for these new cases.

Documentation is also updated to add examples.

Closes #280

DCO 1.1 Signed-off-by: Dan Purdy danjpurdy@gmail.com

### CODE REVIEW

File: tests/rules/no-duplicate-properties.js [hunk 1]
```diff
@@ -28,7 +28,24 @@ describe('no duplicate properties - scss', function () {
         }
       ]
     }, function (data) {
-      lint.assert.equal(3, data.warningCount);
+      lint.assert.equal(5, data.warningCount);
+      done();
+    });
+  });
+
+  it('enforce - [exclude: background]', function (done) {
```
Comment (bgriffith) [inline tests/rules/no-duplicate-properties.js]: Should be `[exclude: background, display]`

File: tests/rules/no-duplicate-properties.js [hunk 2]
```diff
@@ -60,7 +77,24 @@ describe('no duplicate properties - sass', function () {
...
```
Comment (bgriffith) [inline tests/rules/no-duplicate-properties.js]: Should be `[exclude: background, display]`

File: tests/sass/no-duplicate-properties.scss [hunk 3]
```diff
@@ -20,3 +24,16 @@
...
+  background: rgb(200, 54, 54);
+  content: 'dec between background';
+  background: rgba(200, 54, 54, 0.5);
+}
+
+.display {
+  content: 'display';
+  display: flex;
+  display: inline-block;
+
```
Comment (bgriffith) [inline tests/sass/no-duplicate-properties.scss]: Is that whitespace intentional?

Comment (DanPurdy) [inline tests/sass/no-duplicate-properties.scss]: Yeah, to make sure it doesn't error on empty lines.

**Pull request merged**